spec/factories/roles.rb

```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :role, class: Role do
+    association :user, factory: :user
```

Comment:
you don't need to specify the factory here since it's the same as the association